-1

Tested at: 3c416233b66e1b3602a5ef222d72c557d760c875


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #27961 @Dr15Jones: Support concurrent LuminosityBlocks in HLTrigReport
 - #27979 @bsunanda: Run3-hgx220 Convert DDHGCalTBModule algorithm for DD4Hep

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-326d25/2509/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-326d25/2509/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-326d25/2509/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/Vertexing.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/Electron.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/Muon.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/TauPFSpecific.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/PackedCandidate.cc: In member function 'void pat::PackedCandidate::pack(bool)':
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/PackedCandidate.cc:14:74: error: no matching function for call to 'min(ROOT::Math::LorentzVector<ROOT::Math::PtEtaPhiM4D<double> >::Scalar, float)'
   float unpackedPt = std::min(p4_.load()->Pt(), MiniFloatConverter::max());
                                                                          ^
In file included from /cvmfs/cms-ib.cern.ch/nweek-02593/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/include/c++/8.3.1/bits/specfun.h:45,
                 from /cvmfs/cms-ib.cern.ch/nweek-02593/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/include/c++/8.3.1/cmath:1892,
                 from /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/Math/interface/deltaPhi.h:11,
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/Muon.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/PATTauDiscriminator.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/TauPFSpecific.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/TriggerFilter.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/TauJetCorrFactors.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-09-12-2300/src/DataFormats/PatCandidates/src/PackedCandidate.cc:14:22: error: no matching function for call to 'min'
  float unpackedPt = std::min(p4_.load()->Pt(), MiniFloatConverter::max());
                     ^~~~~~~~
/cvmfs/cms-ib.cern.ch/nweek-02593/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../include/c++/8.3.1/bits/algorithmfwd.h:383:5: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('double' vs. 'float')
    min(const _Tp&, const _Tp&);
    ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #27961 @Dr15Jones: Support concurrent LuminosityBlocks in HLTrigReport
 - #27979 @bsunanda: Run3-hgx220 Convert DDHGCalTBModule algorithm for DD4Hep

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-326d25/2509/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-326d25/2509/git-merge-result

--- a/DataFormats/PatCandidates/src/PackedCandidate.cc
+++ b/DataFormats/PatCandidates/src/PackedCandidate.cc
@@ -11,7 +11,7 @@ CovarianceParameterization pat::PackedCandidate::covarianceParameterization_;
 std::once_flag pat::PackedCandidate::covariance_load_flag;
 
 void pat::PackedCandidate::pack(bool unpackAfterwards) {
-  float unpackedPt = std::min(p4_.load()->Pt(), MiniFloatConverter::max());
+  float unpackedPt = std::min<float>(p4_.load()->Pt(), MiniFloatConverter::max());
   packedPt_ = MiniFloatConverter::float32to16(unpackedPt);
   packedEta_ = int16_t(std::round(p4_.load()->Eta() / 6.0f * std::numeric_limits<int16_t>::max()));
   packedPhi_ = int16_t(std::round(p4_.load()->Phi() / 3.2f * std::numeric_limits<int16_t>::max()));

